js_modules/dagster-ui/packages/ui-core/src/app/AppTopNav/AppTopNavContext.tsx

```diff
@@ -1,6 +1,35 @@
-import {atom} from 'recoil';
+import {useCallback, useLayoutEffect} from 'react';
+import {atom, useRecoilState, useSetRecoilState} from 'recoil';
 
 export const isFullScreenAtom = atom<boolean>({
   key: 'isFullScreenAtom',
   default: false,
 });
+
+export const isFullScreenAllowedAtom = atom<boolean>({
+  key: 'isFullScreenAllowedAtom',
+  default: false,
+});
+
+export const useFullscreen = () => {
+  const [isFullScreen, setIsFullScreen] = useRecoilState(isFullScreenAtom);
+  const [isFullScreenEnabled, setIsFullScreenEnabled] = useRecoilState(isFullScreenAllowedAtom);
```

Comment:
I find these slightly confusing as named. I wonder if `renderFullScreen` or something like that would help clarify that one of them is for the user's toggled setting, and one is to actually determine whether the page will be rendered that way.

Reply:
updated the naming